homeassistant/helpers/area_registry.py

```diff
@@ -133,14 +135,20 @@ def _data_to_save(self) -> dict:
 @bind_hass
 async def async_get_registry(hass: HomeAssistantType) -> AreaRegistry:
     """Return area registry instance."""
-    task = hass.data.get(DATA_REGISTRY)
+    reg = hass.data.get(DATA_REGISTRY)
+
+    if not reg:
+        evt = hass.data[DATA_REGISTRY] = Event()
 
-    if task is None:
-        async def _load_reg() -> AreaRegistry:
-            registry = AreaRegistry(hass)
-            await registry.async_load()
-            return registry
+        reg = AreaRegistry(hass)
+        await reg.async_load()
 
-        task = hass.data[DATA_REGISTRY] = hass.async_create_task(_load_reg())
+        hass.data[DATA_REGISTRY] = reg
+        evt.set()
+    elif isinstance(reg, Event):
+        # This shouldn't be possible in production (has been tested)
```

Comment:
This fork will only get reached in testing, since none of the core components do anything with the registries, and the optional line in bootstrap awaits the loading to complete before loading other components. I tested it by replacing it with an exception and it only got thrown in testing.